src/components/AllMonsters.js

```diff
@@ -30,10 +34,13 @@ export default function AllMonsters() {
 
   if (monsterDetails.length !== 0) {
     saveLocally('monsterdetails', monsterDetails)
+    setLoading(false)
   } else if (localStorage.getItem('monsterdetails')) {
     setMonsterDetails(JSON.parse(localStorage.getItem('monsterdetails')))
   }
```

Comment:
This is the reason for your infinite loop. On every render, your component sets your states. Setting a state triggers a render.

Instead, you could move this inside your `useEffect`.